src/components/checkbox/checkbox.tsx

```diff
@@ -97,6 +101,12 @@ export class CheckBox extends React.Component<CheckBoxProps, CheckBoxState> {
         );
     }
 
+    private handleClick = (e: React.SyntheticEvent<HTMLDivElement>) => {
```

Comment:
This might be equally valid, but I prefer putting the type annotation on the handler itself, and then having the event's type inferred from that. i.e. 
`private handleClick: React.EventHandler<React.SyntheticEvent<HTMLDivElement>> = (e): void => {}`

Reply:
After trying it out - this seems to be a matter of code style rather than any validation differences